R/dependencies.R

```diff
@@ -554,3 +557,114 @@ check_yamls_consistent <- function(dep_structure) {
 
   return(invisible(NULL))
 }
+
+
+#' List the external R packages required to be installed
+#'
+#'
+#' @param available_packages (`data.frame`) A dataframe of the format given by
+#'   `as.data.frame(utils::available.packages)`. It is unlikely this default needs to be changed;
+#'   however you need to ensure the `options("repos")` contains the urls of all expected repos
+#'   (e.g. Bioconductor).
+#' @param from_internal_dependencies Vector chosen from `c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")`
+#'   which fields of the DESCRIPTION file of the internal packages should be included. Default:
+#'   `c("Depends", "Imports", "LinkingTo", "Suggests")`
+#' @param from_external_dependencies Vector chosen from `c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")`
+#'   which fields of the DESCRIPTION file of the internal packages should be included. Default:
+#'   `c("Depends", "Imports", "LinkingTo")`
+#' @inheritParams build_check_install
+#' @md
+#' @return A vector of 'external' R packages required to install
+#'   the selected 'internal' packages. This can be used with `remotes::system_requirements`
+#'   to extract the system requirements needed for your packages, see example below.
+#' @examples
+#' \dontrun{
+#'   x <- dependency_table("openpharma/stageddeps.electricity",
+#'     project_type = "repo@@host", feature = "main")
+#'
+#'   # get external package dependencies
+#'   ex_deps <- get_all_external_dependencies(x)
+#'   print(ex_deps)
+#'
+#'   # get system dependencies (in this case there are none)
+#'   unique(unlist(lapply(ex_deps,
```

Comment:
It would be nice to replace this with:

```r
remotes::system_requirements(os = "ubuntu", os_release = "20.04", package = ex_deps)
```

I have created and issue: https://github.com/r-lib/remotes/issues/650